openedx/adg/lms/applications/helpers.py

```diff
@@ -365,67 +365,77 @@ def get_course_card_information(user, courses):
     Returns:
         course_cards_information (list): A list containing dictionaries with status, grade, message and course
         is_any_course_started (bool): Flag indicating if any course has been started yet
-        is_locked (bool): Flag indicating if any course is locked
+        are_courses_completed (bool): Flag indicating if all courses have been passed
     """
     course_cards_information = []
+    locked_course_cards_information = []
+    unlocked_course_cards_information = []
     is_any_course_started = False
-    is_locked = False
+    are_courses_completed = bool(courses)
 
     for course in courses:
-        message = grade = status = ''
+        message = grade = status = pre_req = ''
         pre_requisite_courses = get_prerequisite_courses_display(course)
 
         if pre_requisite_courses:
-            is_locked, status, message = get_information_for_course_with_prerequisites(
-                user, pre_requisite_courses, is_locked, status
+            status, message, pre_req = get_information_for_course_with_prerequisites(
+                user, pre_requisite_courses, status
             )
 
         if status != LOCKED:
             is_any_course_started, status, grade, message = get_information_for_unlocked_course(
                 user, course, is_any_course_started, grade
             )
 
-        course_cards_information.append({
-            'course': course,
-            'status': status,
-            'grade': grade,
-            'message': message
-        })
+        if status != COMPLETED:
+            are_courses_completed = False
 
-    return course_cards_information, is_any_course_started, is_locked
+        course_information = {
+            'course': course, 'status': status, 'grade': grade, 'message': message, 'pre_req': pre_req
+        }
 
+        if pre_requisite_courses:
+            if status == LOCKED:
+                locked_course_cards_information.append(course_information)
+            else:
+                unlocked_course_cards_information.append(course_information)
+        else:
+            course_cards_information.append(course_information)
+
+    course_cards = course_cards_information + unlocked_course_cards_information + locked_course_cards_information
```

Comment:
For future reference, it is a good practice to add a blank line before the return statement for better readability.